SConstruct

```diff
@@ -576,9 +583,17 @@ if selected_platform in platform_list:
             env.Append(CCFLAGS=["/Od"])
     else:
         if env["debug_symbols"]:
-            # Adding dwarf-4 explicitly makes stacktraces work with clang builds,
-            # otherwise addr2line doesn't understand them
-            env.Append(CCFLAGS=["-gdwarf-4"])
+            if selected_platform == "windows" and env_temp["use_llvm"]:
+                # On Windows when using LLVM the -gcodeview creates debug symbols compatible
+                # with Microsoft debuggers.
+                # Additionally -g must be sent to the linker to make it actually produce
+                # a PDB file containing the debug symbols, otherwise they are discarded.
+                env.Append(CCFLAGS=["-gcodeview"])
+                env.Append(LINKFLAGS=["-g"])
```

Comment:
Maybe this should be a separate option, in case someone wants to use clang build with DWARF debug symbols?

Reply:
What would such an option be called? `use_dwarf`? `debug_dwarf`? `dwarf_symbols`?

What I am certain of is that PDB generation should be the default, given the relative amount of tooling for each format on Windows. Using dwarf symbols on Windows would be a niche usecase.

Reply:
Yes, for Windows PDB should be default (and it should also work with existing crash handler - [crash_handler_windows.h](https://github.com/godotengine/godot/blob/master/platform/windows/crash_handler_windows.h), which is currently limited to MSVC).

There are no reasons to select between DWARF-4 and 5, so it probably should be a bool option for Windows/LLVM only (I guess `use_dwarf` to match other `use_*` options).